src/status_im2/contexts/chat/messages/list/view.cljs

```diff
@@ -102,47 +105,107 @@
           [content.deleted/deleted-message message-data context]
           [message/message-with-reactions message-data context keyboard-shown])]))])
 
-(defn messages-list
-  [{:keys [chat-id] :as chat}]
-  (let [context        (rf/sub [:chats/current-chat-message-list-view-context])
-        messages       (rf/sub [:chats/raw-chat-messages-stream chat-id])
-        keyboard-shown (atom false)
-        bottom-space   15]
-    [rn/view
-     {:style {:flex 1}}
-     ;; NOTE: DO NOT use anonymous functions for handlers
-     [rn/flat-list
-      {:key-fn                       list-key-fn
-       :ref                          list-ref
-       :header                       [list-header chat]
-       :footer                       [list-footer chat]
-       :data                         messages
-       :render-data                  {:context        context
-                                      :keyboard-shown keyboard-shown}
-       :render-fn                    render-fn
-       :on-viewable-items-changed    on-viewable-items-changed
-       :on-end-reached               list-on-end-reached
-       :on-scroll-to-index-failed    identity ; don't remove this
-       :content-container-style      {:padding-top    (+ bottom-space 32)
-                                      :padding-bottom 16}
-       :scroll-indicator-insets      {:top bottom-space} ; iOS only
-       :keyboard-dismiss-mode        :interactive
-       :keyboard-should-persist-taps :handled
-       :onMomentumScrollBegin        state/start-scrolling
-       :onMomentumScrollEnd          state/stop-scrolling
-       :scrollEventThrottle          16
-       :on-scroll                    on-scroll
-       ;; TODO https://github.com/facebook/react-native/issues/30034
-       :inverted                     (when platform/ios? true)
-       :style                        (when platform/android? {:scaleY -1})
-       :on-layout                    on-messages-view-layout}]
+(defn calc-shell-position
+  [y]
+  (let [{:keys [input-content-height focused?]} (rf/sub [:chats/current-chat-input])
```

Comment:
I have removed the subscription to make it pure. I tried using React's `useMemo` to memoize the value, but it seems it doesn't work with reagent

Reply:
Oh that's unfortunate to hear @OmarBasem. I believe memoization with Reagent should work fine because we do use it in the codebase, for example in `quo2/components/icon.cljs`. In that namespace we see the usage of Clojure's core function `memoize`. This is the naive memoize, but it works well when we know the args will not vary much. It's important to be sure about this, otherwise the memoization might behave like a memory leak.

If we want to cache (pure) function calls when arguments are more unpredictable, we should use something like https://github.com/clojure/core.memoize. It gives us memoize functions based on time-to-live or least-recently used semantics, etc. But I guess for your use case it's not needed.

Reply:
@ilmotta so I tried using memoize as follows:

created a definition of the function

`(def memoized-calc-shell-position (memoize calc-shell-position))`

Then called it as: `shell-position (memoized-calc-shell-position y-pos chat-input reply edit images)`

But it half-works. When typing new chars, without changing content-height, it keeps calling the function. But when deleting text it uses the memoized value.

Reply:
If I'm not mistaken, there's only one reason if the memoized function keeps being called, and that's if the arguments are not identical. Somehow, args must be different and are triggering the original function call. I can't think of any other reason. So I'd try to inspect args throughout all rendering passes of the component.